presto-record-decoder/src/main/java/com/facebook/presto/decoder/avro/AvroColumnDecoder.java

```diff
@@ -257,12 +274,16 @@ private static void serializeGeneric(BlockBuilder blockBuilder, Object value, Ty
             case BOOLEAN:
                 type.writeBoolean(blockBuilder, (Boolean) value);
                 break;
+            case INTEGER:
             case BIGINT:
-                type.writeLong(blockBuilder, (Long) value);
+                type.writeLong(blockBuilder, ((Number) value).longValue());
```

Comment:
Just wanna confirm, could the value be an Integer or something else here?

Reply:
yes, the value could be Integer, which will fall into 

`type.writeLong(blockBuilder, ((Number) value).longValue());
`
I'm not an expert on this, but my test indicate this line it is compatible with Integer